add deploy to cloudflare docs + fix getting started (#21107)
cloudflare/cloudflare-docs: src/content/docs/workers/deploy-buttons.mdx

```json
{
  "scripts": {
  "build": "astro build", 
  "deploy": "npm run db:migrations:apply && wrangler deploy"
  "db:migrations:apply": "wrangler d1 migrations apply DB_BINDING --remote",
  }
}
```

## Not currently supported 
* **Monorepos**: Cloudflare does not fully support monorepos unless your application is completely isolated within a subdirectory and has no dependencies outside of that subdirectory (for example, [a monorepo of example applications](https://github.com/cloudflare/templates)). If your application relies on dependencies outside its subdirectory, the build will fail. Additionally, if your monorepo contains multiple Workers applications, they will not be deployed together. You must configure a separate Deploy to Cloudflare button for each application. The user will manually create a distinct Workers application for each subdirectory.
* **Pages applications**: Deploy to Cloudflare buttons only support Workers applications today.

> * **Non gitHub.com/gitLab.com repositories**: Source repositories from anything other than gitHub.com and gitLab.com are not supported. Self-hosted versions of GitHub and GitLab are also not supported. 

Reviewer: Hi @bllchmbrs - If it's possible, I think it's ok to omit capitalization suggestions for websites. Thanks!
cc: @kodster28

<Tabs syncKey="DeployButtonSnippet"> <TabItem label="HTML">
```html
<a href="https://deploy.workers.cloudflare.com/?url=<your git repo URL>"><img src="https://deploy.workers.cloudflare.com/button" alt="Deploy to Cloudflare"/></a>
```

</TabItem>

<TabItem label="Markdown">
```md
[![Deploy to Cloudflare](https://deploy.workers.cloudflare.com/button)](https://deploy.workers.cloudflare.com/?url=<git repo URL>)
```

</TabItem>
</Tabs>

If you have already deployed your application using Workers Builds, you can generate a Deploy to Cloudflare button directly from the Cloudflare dashboard by clicking the share button and copying the provided snippet.

<insert image of share icon pop out/>

Once you have your snippet, you can paste this wherever you would like your button to be displayed.

## Automatic Resource Provisioning
If your Worker application requires Cloudflare resources, they will be automatically provisioned as part of the deployment. Currently, supported resources include:
* **Storage**: KV namespaces, D1 databases, R2 buckets, Hyperdrive and Vectorize
* **Compute**: Durable Objects, Workers AI, Containers and Queues

> Cloudflare will read the Wrangler configuration file of your source repo to determine resource requirements for your application. During deployment, Cloudflare will provision any necessary resources and update the Wrangler configuration where applicable for newly created resources (for example, database IDs, namespace IDs). To ensure successful deployment, please make sure your source repository includes default values for resource names, resource IDs and any other properties for each binding.

Reviewer: ```suggestion
Cloudflare will read the Wrangler configuration file of your source repo to determine resource requirements for your application. During deployment, Cloudflare will provision any necessary resources and update the Wrangler configuration where applicable for newly created resources (for example, database IDs and namespace IDs). To ensure successful deployment, please make sure your source repository includes default values for resource names, resource IDs, and any other properties for each binding.
```

> ## Best Practices

Reviewer: ```suggestion
## Best practices
```

> **Configuring Build/Deploy commands**: If you are using custom  `build` and `deploy` scripts in your package.json (e.g. if using a full stack framework or running D1 migrations), Cloudfare will automatically detect and pre-populate the build and deploy fields. Users can choose to modify or accept the custom commands during deployment configuration.

Reviewer: ```suggestion
**Configuring Build/Deploy commands**: If you are using custom  `build` and `deploy` scripts in your package.json (for example, if using a full stack framework or running D1 migrations), Cloudfare will automatically detect and pre-populate the `build` and `deploy` fields. Users can choose to modify or accept the custom commands during deployment configuration.
```

<insert screenshot of build and deploy command UI/>

If no `deploy` script is specified, Cloudflare will preconfigure `npx wrangler deploy` by default. If no `build` script is specified, Cloudflare will leave this field blank.

> ** Running D1 Migrations**: If you would like to run migrations as part of your set up, you can specify this in your `package.json` by running your migrations as part of your `deploy` script. The migration command should reference the binding name rather than the database name to ensure migrations are successful when users specify a database name that is different from that of your source repository. The following is an example of how you can set up the scripts section of your `package.json`:

Reviewer: ```suggestion
**Running D1 Migrations**: If you would like to run migrations as part of your setup, you can specify this in your `package.json` by running your migrations as part of your `deploy` script. The migration command should reference the binding name rather than the database name to ensure migrations are successful when users specify a database name that is different from that of your source repository. The following is an example of how you can set up the scripts section of your `package.json`:
```
